lib/connection.js

```diff
@@ -27,607 +24,606 @@ let PACKET_HANDLERS = null;
-};
+      if (callback) {
+        callback(null, proxy);
+      }
+    };
 
-// Stop sending heartbeat packets
-//
-Connection.prototype.stopHeartbeat = function() {
-  if (this._heartbeatCallbackInstance) {
-    this.clearTimeout(this._heartbeatCallbackInstance);
-    this._heartbeatCallbackInstance = null;
+    this._send(packet);
   }
-};
-
-// Create a JSTP packet
-//   kind - packet kind
-//   target - name of an interface or an application (optional)
-//   verb - action specific for different packet kinds
-//   args - action arguments
-//
-Connection.prototype.createPacket = function(kind, target, verb, args) {
-  const packet = {
-    [kind]: (
-      target ?
-      [this._nextPacketId, target] :
-      [this._nextPacketId]
-    )
-  };
 
-  if (verb) {
-    packet[verb] = args;
+  // Send a ping packet
+  //
+  ping(callback) {
+    const packet = this.createPacket('ping');
+    const packetId = packet.ping[0];
+    this._callbacks[packetId] = callback || common.doNothing;
+    this._send(packet);
   }
 
-  this._nextPacketId += this._packetIdDelta;
-
-  return packet;
-};
+  // Send a pong packet
+  //
+  pong(packetId) {
+    const packet = { pong: [packetId] };
+    this._send(packet);
+  }
 
-// Close the connection
-//
-Connection.prototype.close = function() {
-  this.stopHeartbeat();
-  this.transport.end();
-};
+  // Start sending heartbeat packets
+  //   interval - heartbeat interval in milliseconds
+  //
+  startHeartbeat(interval) {
+    this._heartbeatCallbackInstance =
+      this._heartbeatCallback.bind(this, interval);
+    this._heartbeatCallbackInstance();
+  }
 
-// Set a timeout using timers.enroll()
-//   milliseconds - amount of milliseconds
-//   callback - callback function
-//
-Connection.prototype.setTimeout = function(milliseconds, callback) {
-  timers.enroll(this, milliseconds);
-  timers._unrefActive(this);
-  this.once('_timeout', callback);
-};
+  // Internal function used by startHeartbeat
+  //
+  _heartbeatCallback(interval) {
+    this.transport.send('{}');
+    this.setTimeout(interval, this._heartbeatCallbackInstance);
+  }
 
-// Clear a timeout set with Connection#setTimeout
-//   handler - timer callback to remove
-//
-Connection.prototype.clearTimeout = function(handler) {
-  timers.unenroll(this);
-  this.removeListener('_timeout', handler);
-};
+  // Stop sending heartbeat packets
+  //
+  stopHeartbeat() {
+    if (this._heartbeatCallbackInstance) {
+      this.clearTimeout(this._heartbeatCallbackInstance);
+      this._heartbeatCallbackInstance = null;
+    }
+  }
 
-// timers.enroll() timeout handler
-//
-Connection.prototype._onTimeout = function() {
-  this.emit('_timeout');
-};
+  // Create a JSTP packet
+  //   kind - packet kind
+  //   target - name of an interface or an application (optional)
+  //   verb - action specific for different packet kinds
+  //   args - action arguments
+  //
+  createPacket(kind, target, verb, args) {
```

Comment:
Is there any reason this method is public? If not, can you please prepend an underscore to its name?